server.py

```
@@ -369,6 +484,36 @@ async def _api_chat(req: ChatRequest):
 
+    # --- Live config / SOUL editing ----------------------------------------
+    # GET returns the current config + persona so external clients (the
+    # Gradio drawer is one; curl is another) can mirror what's running.
+    # POST accepts partial edits — pass only the sections you want to
+    # change. Reload is automatic.
+    class ConfigReloadRequest(PydanticBaseModel):
+        config: dict | None = None
+        soul: str | None = None
+
+    @fastapi_app.get("/api/config")
+    async def _api_get_config():
```

Ruff (0.15.10)

[warning] Missing return type annotation for private function `_api_get_config`

(ANN202)